src/Mod/Assembly/App/AssemblyObject.cpp

```diff
@@ -1302,29 +1333,32 @@ void AssemblyObject::getRackPinionMarkers(App::DocumentObject* joint,
         swapJCS(joint);  // make sure that rack is first.
     }
 
-    App::DocumentObject* part1 = getObjFromProp(joint, "Part1");
-    App::DocumentObject* obj1 = getObjFromProp(joint, "Object1");
+    App::DocumentObject* part1 = getMovingPartFromRef(joint, "Reference1");
+    App::DocumentObject* obj1 = getObjFromRef(joint, "Reference1");
     Base::Placement plc1 = getPlacementFromProp(joint, "Placement1");
 
-    App::DocumentObject* part2 = getObjFromProp(joint, "Part2");
-    App::DocumentObject* obj2 = getObjFromProp(joint, "Object2");
+    App::DocumentObject* part2 = getMovingPartFromRef(joint, "Reference2");
```

Comment:
This is no longer used -- is that correct?

Reply:
Indeed you are right!